src/test/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkVersionTest.java

```diff
@@ -71,49 +79,34 @@ public void testConstructor_missingRequiredNumbers() {
   @Test
   public void testConstructor_requiredNumbersOnly() {
     CloudSdkVersion version = new CloudSdkVersion("0.1.0");
-    assertEquals(0, version.getMajorVersion());
-    assertEquals(1, version.getMinorVerion());
-    assertEquals(0, version.getPatchVersion());
```

Comment:
It's still useful for the tests to verify these fields though, right? (and ditto for the rest)

Reply:
That's true, but I don't really want to make the fields package-protected only for testing. Making a method package-protected, I'm a little bit more lenient, OTOH. We have `compareTo()` tests, so we are somewhat covered.

Reply:
Problem is that you don't know that the parsing is correct.  Especially since `toString()` just returns the original version string.  Maybe if `toString()` computed the result then you could compare that.

Reply:
But the test will depend on the implementation details of `toString()`, which is exactly the current situation.

Reply:
With your patch as is, there is no way to verify that the version parsing has correctly identified the version components.  Having `toString()` as a proxy seems ok to me if you don't want to expose `*Version` and friends as `@VisibleForTesting`.

Reply:
I can think of some ways to verify the version parsing indirectly. I'll add some dedicated tests.